File: d01/p1/go/main.go
package main

import (
	"fmt"
	"log"
	"os"
	"strings"
	"strconv"
)

func	main() {
	
	if len(os.Args) != 2 {
		log.Fatalln("Program requires one argument")
	}
	var pw int
	dial := 50
	input, err := os.ReadFile(os.Args[1])
	if err != nil {
		log.Fatalln("Error reading from file", err)
	}
	moves := strings.Split(string(input), "\n")

	fmt.Println(" Dial: ", dial)
	for i, instruction := range moves {
		
		if instruction == "" {
			continue
		}
		if instruction[0] == 'L' {
			clicks, err := strconv.Atoi(instruction[1:])
			if err != nil {
				log.Fatalln("Error in Atoi", err)
			}
			dial = (dial - clicks) % 100
			for dial < 0 {
				dial += 100
			}
		} else if instruction[0] == 'R' {
			clicks, err:= strconv.Atoi(instruction[1:])
			if err != nil {
				log.Fatalln("Error in Atoi", err)
			}
			dial = (dial + clicks) % 100
		}
		fmt.Println("Instruction[", i, "]: ", instruction, "  Dial: ", dial)
		if dial == 0 {
			pw++
		}
	}
	fmt.Println("The password is", pw)
}

File: d01/p1/C/main.c
#include <fcntl.h>
#include <unistd.h>

int	main(int ac, char **av)
{
	if (ac != 2)
		return (write(2, "Program requires one argument\n", 30), 1);
	char input[];
	int	i = 0;
	int	j = 0;
	int	pw = 0;
	int	dial = 50;
	int	fd = open(av[1], O_RDONLY);	
	int	bytes = read(fd, input, sizeof(buff));
	if (bytes < 0)
		return (write(2, "Error reading file\n", 19), 1);
	char	**moves = ft_split(input, '\n');
	if (!moves)
		return (write(2, "Error in split's malloc\n", 24), 1);
	

	while (moves[i])
	{
		int clicks = atoi(&moves[i][1]);
		if (moves[i][0] == 'L')
		{
		
		}
		else if (move[i][0] == 'R')
		{
		
		}
		if (dial == 0)
			pw++;
	}

	return (0)
}
